src/DataTable/index.jsx

```diff
@@ -1,76 +1,93 @@
-import React from 'react'
-
-import Pagination from './Pagination'
-import Row from './Row'
-import Search from './Search'
+import React, { Fragment } from "react";
+import { getTotalNumberOfPages, search } from "./logic";
+import PaginationBox from "./PaginationBox";
+import Table from "./Table";
+import SearchBox from "./SearchBox";
 
 class DataTable extends React.Component {
   state = {
-    rows: this.props.rows,
-    currentPageNumber: 0,
-    totalNumberOfPages: this.calculateTotalNumberOfPages(this.props.rows)
-  }
-
-  static defaultProps = {
-    rowsPerPage: 40
-  }
+    currentPageNumber: 1,
+    rowsPerPage: 5,
+    searchText: "",
+    rowsFound: [],
+  };
 
-  calculateTotalNumberOfPages(rows) {
-    const { rowsPerPage } = this.props
-    if (rowsPerPage == 0) return 0
-    return Math.ceil(rows.length / rowsPerPage)
+  componentDidMount() {
+    const rows = this.props.rows;
+    this.setState(() => ({
+      rowsFound: search(rows, this.state.searchText),
+    }));
   }
 
-  search(event) {
-    const { rows } = this.props
-    const text = event.target.value
-    let rowsFound = rows
-
-    if (text) {
-      rowsFound = rows.filter((row) => {
-        return row.name1.toLowerCase().search(text.toLowerCase()) > -1 ||
-         (row.email && row.email.toLowerCase().search(text.toLowerCase()) > -1)
-      })
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.searchText !== prevState.searchText) {
+      const rows = this.props.rows;
+      this.setState({
+        rowsFound: search(rows, this.state.searchText),
+      });
+    }
+    if (this.state.rowsPerPage !== prevState.rowsPerPage) {
+      const totalNumberOfPages = getTotalNumberOfPages(
+        this.state.rowsFound,
+        this.state.rowsPerPage
+      );
+      if (this.state.currentPageNumber > totalNumberOfPages) {
+        this.setState({
+          currentPageNumber: 1,
+        });
+      }
     }
-
-    this.setState({
-      rows: rowsFound,
```

Comment:
`rows: rowsFound,` I think it is confusing, why not just call it `rowsFound` and code can be shorter